client/src/utils/fight/hypnotise.ts

```diff
@@ -27,60 +27,182 @@ const hypnotise = async (
+  const update = (delta: number) => {
+    // Reset graphics
+    spiral.clear();
+
+    // Thickness start
+    let thickness = 0;
+    const centerX = app.screen.width / 2;
+    const centerY = app.screen.height / 2;
+
+    // Number of turns
+    const turns = 12;
+
+    // Draw the spiral
+    for (let i = 0; i < turns * Math.PI * 2; i += 0.1) {
+      thickness += thicknessDelta;
+      spiral.lineStyle(thickness, 0xffffff);
+      const radius = i * radiusIncrement;
+      const x = centerX + radius * Math.cos(i + angle);
+      const y = centerY + radius * Math.sin(i + angle);
+      const moveX = centerX + radius * Math.cos(i + angle - 0.05);
+      const moveY = centerY + radius * Math.sin(i + angle - 0.05);
+
+      if (i === 0) {
+        spiral.moveTo(x, y);
+      } else {
+        spiral.lineTo(x, y);
+        spiral.moveTo(moveX, moveY);
+      }
+    }
+    // Update params
+    angle -= 0.25 * speed.current * delta;
+    radiusIncrement += 0.015 * speed.current * delta;
+    thicknessDelta -= 0.00048 * speed.current * delta;
+    // Stop
+    if (stopDraw) app.ticker.remove(update);
+  };
+
+  app.ticker.add(update);
+
+  // Fade in spiral
+  await Tweener.add({
+    target: spiralContainer,
+    duration: 0.05 / speed.current,
     ease: Easing.linear,
   }, {
-    width: 200,
-    height: 200,
-  }));
+    alpha: 1,
+  });
 
-  // Wait for all animations to complete
-  await Promise.all(animations);
+  // Fade out inverted container
+  await Tweener.add({
+    target: invertedContainer,
+    duration: 1 / speed.current,
+    ease: Easing.linear,
+  }, {
+    alpha: 0,
+  }).then(() => {
+    // Remove from stage
+    app.stage.removeChild(invertedContainer);
+  });
+
+  // Fade out spiral container to 0.3
+  await Tweener.add({
+    target: spiralContainer,
+    duration: 1.2 / speed.current,
+    ease: Easing.linear,
+  }, {
+    alpha: 0.3,
+  });
+
+  // Reset hypnosis filter
+  hypnosisFilter.negative(true);
+  hypnosisFilter.hue(-75, true);
+
+  // Slight flash animation
+  let brightness = 10;
+  const flash = () => {
+    brightness -= 1;
+    hypnosisFilter.brightness(brightness, false);
+    if (brightness <= 1 && app.stage.filters) {
+      // Flash animation end
+      app.ticker.remove(flash);
+      // Remove hypnosis filter
+      const index = app.stage.filters?.findIndex((f) => f === hypnosisFilter);
+      if (index !== undefined && index !== -1) {
+        app.stage.filters?.splice(index, 1);
+      }
+    }
+  };
+
+  // Start flash animation
+  app.ticker.add(flash);
+
+  // Fade out spiral container
+  void Tweener.add({
+    target: spiralContainer,
+    duration: 0.5 / speed.current,
+    ease: Easing.linear,
+  }, {
+    alpha: 0,
+  }).then(() => {
+    // End spiral animation
+    stopDraw = true;
+    // Remove from stage
+    app.stage.removeChild(spiralContainer);
+  });
 
-  // Destroy wave
-  wave.destroy();
+  // Set animation to idle
+  brute.animation.setAnimation('idle');
```

Comment:
This animation change is too sudden, it feels like there is a cut in the fight playback.
A solution would be to add the possibility to play animations in reverse in FighterHolder, and use that to play the win animation in reverse until the event `win:start` trigger, then set the animation to idle